_src/pattern-library/js/select-replace.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Select menu replacement
+ * Handles functionality for the replacement select menus, which
+ * allows us to style them to our liking. Accessibility is main-
+ * tained via use of the default select control. The replacement 
+ * control is not visible to screen readers.
+ *
+ * Choosing an option in the replacement menu also updates the
+ * default select menu thus maintaining accessibility.
+ */
+
+$(function() {
+
+    var CustomSelectReplacement = {
+
+        vars: {
+            replaced:       $('.replace-select'),
+            replacedClass:  'is-replaced is-transparent',
+            customClass:    'wrapper-custom-select',
+            wrapperClass:   'wrapper-replace-select',
+            valueClass:     'replace-value',
+            iconClass:      'icon-caret-down',
+            hoverClass:     'is-hover'
+        },
+
+        init: function() {
+            this.replaceFoundSelects();
+            this.listenForSelectClick();
+            this.listenForSelectEvents();
+        },
+
+        replaceFoundSelects: function() {
+            var variables = this.vars;
+
+            if (variables.replaced.length) {
+
+                variables.replaced.each(function(index, el) {
+                    var $el = $(el),
+                        statuses = '';
+
+                    if ($el.hasClass('has-success')) {
+                        statuses += 'has-success';
+                    }
+
+                    if ($el.hasClass('has-error')) {
+                        statuses += 'has-error';
+                    }
+
+                    if ($el.is(':disabled')) {
+                        statuses += 'is-disabled';
```

Comment:
There are no spaces between the classes if multiple are added. I would suggest making statuses a classList array, pushing the classes here and then outputting statuses.join(' ') to add the classes to the DOM element.